iOSClient/AppDelegate.swift

```diff
@@ -30,7 +30,7 @@ import Firebase
 import IHProgressHUD
 
 @UIApplicationMain
-class AppDelegate: UIResponder, UIApplicationDelegate, UNUserNotificationCenterDelegate, TOPasscodeViewControllerDelegate, NCAccountRequestDelegate, NCViewCertificateDetailsDelegate {
+class AppDelegate: UIResponder, UIApplicationDelegate, UNUserNotificationCenterDelegate, TOPasscodeViewControllerDelegate, NCAccountRequestDelegate, NCViewCertificateDetailsDelegate, NCUserBaseUrl {
 
```

Comment:
NCUserBaseUrl ?

Reply:
Yea, I'm not happy with the naming as well.. if you have a better idea let me know..

Reply:
I don't undestand because you use a new class NCUserBaseUrl for a simple creation of name of avatar ...  user + "-" + (URL(string: urlBase)?.host ?? "")

Reply:
It's not a class it's a protocol. It just generalises how the user-url-id is created, for all types that can do so.

Reply:
ja ja protocol (same) then we can, then, create a new file as NCCellProtocol.swift

Reply:
of name .... NCUserFile -> var userFile: String , sounds good for you ?

Reply:
> ja ja protocol (same)

Actually, they're not and if you haven't seen it I highly recommend the "Crusty talk" to you..

About the name, I was more thinking of the protocol as a _capability_, but the more I think about it it's actually just means that the type has (or is part of) a user and a server url. So it makes sense as it is (to me). And I think that since it combines the user and url, they should be reflected in the name as well.

Reply:
urlBase was the name of old (url) ownCloud driver objc of several year ago ... this project is many many years old and now seems simple but several year ago was a massacre... and many new technologies weren't there :)

Reply:
a million things can be improved but let's take it easy ...